service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CarePlanMapper.java

```diff
@@ -0,0 +1,129 @@
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component3;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01StructuredBody;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanStatus.ACTIVE;
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanIntent.PLAN;
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+import static uk.nhs.adaptors.oneoneone.cda.report.util.ReferenceUtil.ofTypes;
+
+@Component
+@AllArgsConstructor
+public class CarePlanMapper {
+    private final String SNOMED                   = "2.16.840.1.113883.2.1.3.2.4.15";
+    private final String INFORMATION_ADVICE_GIVEN = "1052951000000105";
+
+    private List<CarePlan> carePlansCreated = new ArrayList<>();
+
+    public List<Reference> mapCarePlan(POCDMT000002UK01ClinicalDocument1 clinicalDocument,
+                                       Encounter encounter) {
+
+        POCDMT000002UK01StructuredBody structuredBody = SectionUtil.getStructuredBody(clinicalDocument);
+
+        return Arrays.stream(structuredBody.getComponentArray())
+                .map(POCDMT000002UK01Component3::getSection)
+                .map(this::findCarePlanSections)
+                .flatMap(List::stream)
+                .map(section -> createCarePlanFromSection(section, encounter))
+                .map(cp -> create(cp))
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public CarePlan createCarePlanFromSection(POCDMT000002UK01Section cpSection, Encounter encounter) {
+        CarePlan carePlan = new CarePlan();
+        carePlan.setIdElement(newRandomUuid());
+
+        if (cpSection.isSetLanguageCode()) {
+            carePlan.setLanguage(NodeUtil.getNodeValueString(cpSection.getLanguageCode()));
+        }
+        carePlan.setTitle(NodeUtil.getNodeValueString(cpSection.getTitle()));
+        String cpTextContent = NodeUtil.getNodeValueString(cpSection.getText().getContentArray(0));
+
+        Narrative narrative = new Narrative();
+        narrative.setDivAsString(cpTextContent);
+        carePlan.setText(narrative);
+        carePlan.setDescription(cpTextContent);
+        carePlan
+            .setIntent(PLAN)
+            .setSubject(encounter.getSubject())
+            .setSubjectTarget(encounter.getSubjectTarget())
+            .setStatus(ACTIVE)
+            .setContextTarget(encounter)
+            .setContext(new Reference(encounter))
+            .setPeriod(encounter.getPeriod());
+
+        getSupportingInfoFromITK(carePlan, cpSection);
+
+        return carePlan;
+    }
+
+    private void getSupportingInfoFromITK(CarePlan carePlan, POCDMT000002UK01Section carePlanSection) {
```

Comment:
parameter carePlanSection seems to be unused int the method

Reply:
Removed - Dawid sported this also.